simplq/src/components/Layout/index.jsx

```diff
@@ -1,14 +1,16 @@
 import React from 'react';
+import { useAuth0 } from '@auth0/auth0-react';
 import Routes from './Routes';
 import Footer from '../common/Footer';
 import styles from './layout.module.scss';
+import LoadingIndicator from '../common/LoadingIndicator';
 
 function Layout() {
+  const { isLoading } = useAuth0();
+
   return (
     <div className={styles['box']}>
-      <div className={styles['content']}>
-        <Routes />
-      </div>
+      <div className={styles['content']}>{isLoading ? <LoadingIndicator /> : <Routes />}</div>
```

Comment:
I want to load the main site content only after auth is loaded. The entire page will have a loading indicator for a moment on page load, but this eases a lot of things

Reply:
I don't really understand the problem you've solved here, could you explain?

Reply:
Auth0 takes a small amount of time at page load to initialise. If we make a API call before this, it would result in auth failure from the backend API. There are other places, for example, we have to wait for it to load before deciding if we should render the login or logout button. Same with the temporary queue warning ribbon.

So it looked visually bad, with too much spinners and changes on page refresh. Also had to handle it in code everywhere.

So before we load the main page section, I thought of waiting till Auth0 "is loaded".

Reply:
Okay, got it. So you essentially made auth a sync call. That makes sense.

Reply:
Basically, I am waiting for auth to finish (at page load auth0 would get a fresh token from their server if user is already logged in, else it set itself to "not logged in") before rendering the content.